modules/apps/headless-apio/media-object/media-object-apio-test/src/testIntegration/java/com/liferay/media/object/apio/internal/architect/resource/test/DefaultMediaObjectNestedCollectionResourceTest.java

```diff
@@ -91,22 +110,128 @@ public void testAddMediaObject() throws Exception {
 			"My media object testAddMediaObject", fileEntry.getTitle());
 	}
 
+	@Sync
 	@Test
-	public void testGetFileEntryPreviewURL() throws Exception {
-		String content = RandomTestUtil.randomString(10);
+	public void testGetAdaptiveMedias() throws Exception {
+		String fileName = "image.jpeg";
+
+		byte[] bytes = FileUtil.getBytes(
+			getClass(),
+			"/com/liferay/media/object/apio/internal/architect/resource/test/" +
+				fileName);
+
+		BinaryFile binaryFile = new BinaryFile(
+			new ByteArrayInputStream(bytes), (long)bytes.length, "image/jpeg",
+			fileName);
 
 		FileEntry fileEntry = _addFileEntry(
 			_group.getGroupId(),
 			new MediaObjectImpl(
-				new BinaryFile(
-					new ByteArrayInputStream(content.getBytes(UTF_8)), 0L,
-					"application/octet-stream"),
-				"My media object testGetFileEntryPreviewURL", null, null,
+				binaryFile, "My media object testGetAdaptiveMedia", null, null,
 				null));
 
+		List<AdaptiveMedia<AMImageProcessor>> adaptiveMediaList =
+			_getAdaptiveMedias(fileEntry);
+
+		Assert.assertEquals(
+			adaptiveMediaList.toString(), 2, adaptiveMediaList.size());
+
+		AdaptiveMedia<AMImageProcessor> adaptiveMedia1000px =
+			adaptiveMediaList.get(0);
+
+		URI contentURL1000px = adaptiveMedia1000px.getURI();
+
+		URL url1000px = new URL(TestPropsValues.PORTAL_URL + contentURL1000px);
```

Comment:
Is contentURL1000px only used here? Replace by adaptiveMedia1000px.getURI()

Reply:
👍